pkg/deploy/util/util.go

```diff
@@ -233,6 +235,66 @@ func MakeDeployment(config *deployapi.DeploymentConfig, codec runtime.Codec) (*a
 	return deployment, nil
 }
 
+// GetReplicaCountForDeployments returns the sum of all replicas for the
+// given deployments.
+func GetReplicaCountForDeployments(deployments []api.ReplicationController) int {
+	totalReplicaCount := 0
+	for _, deployment := range deployments {
+		totalReplicaCount += deployment.Spec.Replicas
+	}
+	return totalReplicaCount
+}
+
+// GetActualReplicaCountForDeployments returns the sum of actual replicas for the
+// given deployments.
+func GetActualReplicaCountForDeployments(deployments []api.ReplicationController) int {
+	totalReplicaCount := 0
+	for _, deployment := range deployments {
+		totalReplicaCount += deployment.Status.Replicas
+	}
+	return totalReplicaCount
+}
+
+// GetAvailablePodsForDeployments returns the number of available pods corresponding
+// to the given deployments.
+func GetAvailablePodsForDeployments(c kclient.Interface, deployments []api.ReplicationController, minReadySeconds int) (int, error) {
+	allPods, err := GetPodsForDeployments(c, deployments)
+	if err != nil {
+		return 0, err
+	}
+	return getReadyPodsCount(allPods, minReadySeconds), nil
+}
+
+func getReadyPodsCount(pods []api.Pod, minReadySeconds int) int {
+	readyPodCount := 0
+	for _, pod := range pods {
+		if kdeplutil.IsPodAvailable(&pod, minReadySeconds) {
+			readyPodCount++
+		}
+	}
+	return readyPodCount
+}
+
+// GetPodsForDeployments returns all the pods for the provided deployments.
+func GetPodsForDeployments(c kclient.Interface, deployments []api.ReplicationController) ([]api.Pod, error) {
+	allPods := map[string]api.Pod{}
+	for _, deployment := range deployments {
+		options := api.ListOptions{LabelSelector: labels.Set(deployment.Spec.Selector).AsSelector()}
+		podList, err := c.Pods(deployment.ObjectMeta.Namespace).List(options)
```

Comment:
This helper is called for calculating the available pods for a deploymentconfig so all deployments and pods will be at the same namespace. I will add a comment clarifying it.